src/extension.ts

```diff
@@ -246,10 +255,18 @@ export async function attach(
                     DevToolsPanel.createOrShow(context, telemetryReporter, selection.detail, runtimeConfig);
                 }
             }
-        } else {
-            telemetryReporter.sendTelemetryEvent('command/attach/error/no_json_array', telemetryProps);
         }
     } while (useRetry && Date.now() - startTime < timeout);
+
+    // If there is no after the timeout then throw an exception
+    if (!responseArray) {
```

Comment:
this probably needs to now be if responseArray.length === 0, since you updated the getListOfTargets to return an empty array instead of undefined on failure? unless [] is falsy. I can't ever remember

Reply:
good catch!